driver/src/price_finding/price_finder_interface.rs

```diff
@@ -20,6 +22,38 @@ impl Default for Fee {
     }
 }
 
+#[derive(Clone, Copy, PartialEq)]
+pub enum OptimizationModel {
+    NAIVE,
+    MIP,
+    NLP,
+}
+
+impl From<String> for OptimizationModel {
+    fn from(optimization_model_string: String) -> OptimizationModel {
+        match optimization_model_string.to_lowercase().as_str() {
+            "mip" => OptimizationModel::MIP,
+            "nlp" => OptimizationModel::NLP,
+            // the naive solver is the standard solver.
+            _ => OptimizationModel::NAIVE,
+        }
+    }
+}
+
+impl OptimizationModel {
+    pub fn to_args(self) -> Result<&'static str, PriceFindingError> {
+        match self {
+            OptimizationModel::MIP => Ok(&"mip"),
+            OptimizationModel::NLP => Ok(&"nlp"),
+            // to_args is currently only used for the optimizations models mip and nlp
+            _ => Err(PriceFindingError::new(
```

Comment:
Small but important note:

It's usually better to spell out all options (when they are finite). This way the compiler will yell at you whenever you create a new option and force you to think about how that new option should be handled in all callsites that are important. If you use the default fallback, the compiler will not warn you and you might end up with weird run time errors later.